command/job_start.go

```diff
@@ -162,3 +190,24 @@ func (c *JobStartCommand) Run(args []string) int {
 	mon := newMonitor(c.Ui, client, length)
 	return mon.monitor(resp.EvalID)
 }
+
+func parseFromSubmission(sub *api.JobSubmission) (*api.Job, error) {
```

Comment:
If the previous version of the job was submitted directly via the HTTP API rather than from the CLI/UI, it won't have an associated submission object and `sub` will be nil and this function will panic. I see three possible ways to handle that:
1. Return an error that we just can't safely do `job start` for a job with scaling policies.
2. Enable all the policies unconditionally (maybe with a warning)?
3. Leave the policies disabled and provide a warning that we weren't able to safely determine whether we can re-enable policies?

I'm slightly in favor of option (3) just because it lets the user move forward without silent misbehavior, and only impacts users who are using the HTTP API directly.

Reply:
Im going with the third one as well, I tried looking at how does the UI handle this, but it just displays the latest stored job, and that won't work here.